internal/pkg/watch/watch.go

```diff
@@ -124,3 +129,57 @@ func (w *Watch) Emit(message interface{}) {
 		}
 	}
 }
+
+func (w *Watch) parseJSON(body []byte) (map[string]interface{}, error) {
+	var jsonResult map[string]interface{}
+	err := json.Unmarshal(body, &jsonResult)
+	if err != nil {
+		return nil, err
+	}
+
+	return jsonResult, nil
+}
+
+func (w *Watch) emitEvents(evs map[string]model.FromContext, body map[string]interface{}) {
+	// search for & emit events
+	for _, ev := range evs {
+		newev, err := ev.New(body, timesync.Now())
+		if err != nil {
+			zap.S().Warnf("event construction error %v", err)
+
+			continue
+		}
+
+		if newev == nil {
+			// nothing to do
+			continue
+		}
+
+		if err := emit.Ev(w, newev); err != nil {
+			zap.S().Error(err)
+
+			continue
+		}
+
+		// stop if at least one event was emitted
+		// for the same buffer
+		break
+	}
+}
+
+func (w *Watch) emitLogStatusEvent(name string) {
+	ctx := map[string]interface{}{
+		model.NodeIDKey:      global.BlockchainNode.NodeID(),
+		model.NodeTypeKey:    global.BlockchainNode.NodeRole(),
+		model.NodeVersionKey: global.BlockchainNode.NodeVersion(),
```

Comment:
There are cases when containers are undiscovered and these fields are empty. Perhaps we don't want to have empty context values arrive to our platform?

Reply:
Addressed in 264b24eec14d.